.github/workflows/docker-build.yml

```diff
@@ -1,15 +1,51 @@
 name: Docker Image CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
 
-jobs:
+permissions:
+  contents: read
+  packages: write
 
+jobs:
   docker-build:
-
     runs-on: ubuntu-latest
-
     steps:
     - name: Checkout
       uses: actions/checkout@v4
-    - name: Build the Docker image
-      run: docker build . --file Dockerfile --tag spellcheck:$(date +%s)
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Login to GitHub Registry
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Metadata for the image
+      id: meta
+      uses: docker/metadata-action@v5
```

Comment:
Please add the SHA sum for the action version with the human readable version numbers as a appended comment as per the GitHub recommendations:

- https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions